httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/PoolingAsyncClientConnectionManager.java

```diff
@@ -183,6 +184,12 @@ public void execute(final Boolean result) {
 
                                 })));
                             } else {
+                                if (!connection.isOpen()) {
```

Comment:
The stale connection check (as an attempt to read from the socket) is only applicable to blocking connections.